File: F#/F#/dequeue.fs
module FSharp.Dequeue

type Dequeue<'T>() =
    let data = ResizeArray<'T>()
    
    member this.addFront item = 
        data.Insert(0, item)
    
    member this.addRear item =
        data.Add item
    
    member this.removeRear () = 
        let res = data.[data.Count - 1]
        data.RemoveAt(data.Count - 1)
        res

    member this.removeFront () =
        let res = data.[0]
        data.RemoveAt(0)
        res
    
    member this.size () = data.Count

File: Python/dequeue.py
__author__ = 'Daniel'


class Dequeue:
    def __init__(self):
        self.data = []

    def add_front(self, item):
        self.data.insert(0, item)

    def add_rear(self, item):
        self.data.append(item)

    def remove_front(self):
        return self.data.pop(0)

    def remove_rear(self):
        return self.data.pop()

    def size(self):
        return len(self.data)